From 21d56f9c91a0a1b55cc7e5933974d6afcef7b001 Mon Sep 17 00:00:00 2001
From: Kunkun Jiang <jiangkunkun@huawei.com>
Date: Mon, 19 Jul 2021 11:20:58 +0800
Subject: [PATCH] iommu/io-pgtable-arm: Remove the limitation on the page table
Patch-mainline: Not yet, from openEuler
References: bsn#22
openEuler-commit: 21d56f9c91a0a1b55cc7e5933974d6afcef7b001

 format of sync/clear_dirty_log()

virt inclusion
category: feature
bugzilla: https://gitee.com/openeuler/kernel/issues/I401IF
CVE: NA

------------------------------

sync/clear_dirty_log() support other Arm types of page table
formats, except 32/64-bit stage 1 format. So it'd better to
remove the limination.

Signed-off-by: Kunkun Jiang <jiangkunkun@huawei.com>
Reviewed-by: Keqian Zhu <zhukeqian1@huawei.com>
Signed-off-by: Zheng Zengkai <zhengzengkai@huawei.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 drivers/iommu/io-pgtable-arm.c | 8 --------
 1 file changed, 8 deletions(-)

diff --git a/drivers/iommu/io-pgtable-arm.c b/drivers/iommu/io-pgtable-arm.c
index 836639aa86bd..b878acad4104 100644
--- a/drivers/iommu/io-pgtable-arm.c
+++ b/drivers/iommu/io-pgtable-arm.c
@@ -979,10 +979,6 @@ static int arm_lpae_sync_dirty_log(struct io_pgtable_ops *ops,
 	if (WARN_ON(iaext))
 		return -EINVAL;
 
-	if (data->iop.fmt != ARM_64_LPAE_S1 &&
-	    data->iop.fmt != ARM_32_LPAE_S1)
-		return -EINVAL;
-
 	return __arm_lpae_sync_dirty_log(data, iova, size, lvl, ptep,
 					 bitmap, base_iova, bitmap_pgshift);
 }
@@ -1075,10 +1071,6 @@ static int arm_lpae_clear_dirty_log(struct io_pgtable_ops *ops,
 	if (WARN_ON(iaext))
 		return -EINVAL;
 
-	if (data->iop.fmt != ARM_64_LPAE_S1 &&
-	    data->iop.fmt != ARM_32_LPAE_S1)
-		return -EINVAL;
-
 	return __arm_lpae_clear_dirty_log(data, iova, size, lvl, ptep,
 			bitmap, base_iova, bitmap_pgshift);
 }
-- 
2.26.2

